src/component/form/AttribElement.tsx

```diff
@@ -1,37 +1,56 @@
+import * as isEqual from 'fast-deep-equal';
 import * as React from 'react';
 import { connect } from 'react-redux';
 import { v4 as generateUUID } from 'uuid';
-
 import { AttributeType, ResultType } from '../../flowTypes';
-import { AppState, SearchResult, UpdateContactFields, updateContactFields } from '../../store';
+import { AppState, SearchResult } from '../../store';
 import { getSelectClass, propertyExists } from '../../utils';
 import SelectSearch from '../SelectSearch';
 import FormElement, { FormElementProps } from './FormElement';
 
 interface AttribElementPassedProps extends FormElementProps {
     initial: SearchResult;
-    endpoint?: string;
+    endpoint: string;
     add?: boolean;
     placeholder?: string;
     searchPromptText?: string;
+    helpText?: string;
 }
 
 interface AttribElementStoreProps {
     contactFields: SearchResult[];
-    updateContactFields: UpdateContactFields;
 }
 
 export type AttribElementProps = AttribElementPassedProps & AttribElementStoreProps;
 
-interface AttribState {
+interface AttribElementState {
     attribute: SearchResult;
     errors: string[];
 }
 
 export const PLACEHOLDER = 'Enter the name of an existing attribute or create a new one';
 export const NOT_FOUND = 'Invalid attribute name';
+export const VALID_FIELD = /^[a-z0-9-][a-z0-9- ]*$/;
+export const CREATE_PROMPT = 'New attribute: ';
+
+export const attribExists = (newOptName: string, options: SearchResult[]) => {
+    const loweredNTrimmed = newOptName.toLowerCase().trim();
+    if (options.length) {
+        for (const { name } of options) {
+            if (name.toLowerCase().trim() === loweredNTrimmed) {
+                return true;
+            }
+        }
+    }
+    return false;
+};
+
+export const fieldNameValid = (name: string = '') => {
+    const lowered = name.toLowerCase();
+    return lowered.length > 0 && lowered.length <= 36 && VALID_FIELD.test(lowered);
```

Comment:
We should have global config somewhere, been putting them in utils (which is where i export the ones from scss land). Would be nice to not see a 36 buried in our code.

Reply:
Cleaned this up a bit more here https://github.com/nyaruka/floweditor/pull/252/commits/e1a3e474987cbe9a879d915fff4f9ef81c478eb7. 

I'm fine with them living in utils for now, but certainly not opposed to moving them to another location if you know where you'd like them to be. Not sure what to call them/where in the repo would make the most sense outside of utils.